daffodil-test/src/test/resources/org/apache/daffodil/section07/variables/variables.tdml

```diff
@@ -1061,4 +1061,67 @@
     </tdml:errors>
   </tdml:parserTestCase>
 
+
+  <tdml:defineSchema name="logical_default_values">
+    
+    <xs:include schemaLocation="org/apache/daffodil/xsd/DFDLGeneralFormat.dfdl.xsd"/>
+    <dfdl:format ref="ex:GeneralFormat" />
+
+    <dfdl:defineVariable name="float" type="xs:float" defaultValue="-INF" />
+    <dfdl:defineVariable name="double" type="xs:double" defaultValue="1.0e-5" />
+    <dfdl:defineVariable name="bool" type="xs:boolean" defaultValue="false" />
```

Comment:
Are 'false' and 'true' actually values?  Does the DFDL spec really allow defaultValue="false" for booleans?
I'm not against this. I'm just not sure where the DFDL spec blesses a syntax for boolean values.

Maybe this is missing from the DFDL spec? 

I know there is no syntax for specifying nil, for example. There's no way to have an expression on say, dfdl:inputValueCalc cause a nillable element to be nilled. So there are definitely limitations on the expression language's capabilities. 

In actual expressions, you have to use fn:true() and fn:false(). The literal value syntax of "true" and "false" isn't allowed. 

I know for a boolean element the XSD attribute default="false" would be allowed, but that's not the same language of literals as DFDL string literals. E.g., in XSD attribute default="....", you can't use DFDL entities.

Reply:
This came up in a review on changes needed to get ISO8583 to work with Daffodil:

https://github.com/DFDLSchemas/ISO8583/pull/2#pullrequestreview-183819263

You'll need to click "show resolved" to show the conversation about the defaultValue="false".

Section 7.7 of the spec says that a logical value is allowed as the defaultValue. I'm not sure the spec is clear on what is meant by a "logical value", but Steve Hanson did state in the review that "false" should be allowed.

For this implementation, this change requires that the logical type be parsable by PrimType.fromXMLString. In the boolean case that's just Scala's string.toBoolean method, which means it will accept "true" or "false" ignoring case. Note that PrimeType.fromXMLString is what we use for converting XML Infoset strings to the internal types, so in theory this allows anything that XSD would allow for a type. Seems like a reasonable interpretation of logical type. So for example, for xs:hexBinary you could do ``defaultValue="DEADBEEF"``.

Regarding inputValueCalc, this change would potentially allow something like dfd:inputValueCalc="false" to work if the type was xs:boolean, but I don't think that's new behavior. That would have worked before this change except that there are other checks  that ensure that inputValueCalc is an actual expression, and the above wouldn't be allowed.